### PR TITLE
Replace usages of innerHTML with innerText or by appending elements

### PR DESCRIPTION
**Pull request by vezwork**
Fixes part 4 _(Warnings: Use of .innerHTML with dynamic content.)_ of #65.

## Changes

- Replace usages of `element.innerHTML` either with:
  - `.innerText` (when the element only contains text) or
  - using new HTML helper functions to programmatically create and insert html elements into the element (when the element contains other elements). These HTML helper function are located at the bottom of `popup.js`.
- Renames `getSentAssetsMapAsString` to `getSentAssetsMapAsEls` and moves it from `main.js` to `popup.js` (since it now uses the HTML helper functions).
- Add css styling to `a` and `p` elements in top site detail to make up for styling previously added by `.innerHTML`.

## Testing Notes

- Test links and text in Chrome, Edge, and Firefox.
- No functionality or appearances should have changed.
- Please ensure text and links display correctly in the following areas and circumstances:
  - when no sites have been visited, on the main page,
  - when monetized sites have been visited without a provider, on the main page,
  - when monetized sites have been visited with a provider, on the main page,
  - top site detail without a provider, and
  - top site detail with a provider.

### CODE REVIEW

**Comment by vezwork** [comment]
I have tested locally without a provider, and it is working as expected✅.

**Comment by vezwork** [comment]
I have tested locally with Coil, and it is working as expected✅.

File: src/popup/popup.js [hunk 1]
```diff
@@ -74,38 +74,53 @@ async function getStats() {
...
-		// The user has not visisted a monetized site yet
+		// The user has not visited a monetized site yet
 		document.getElementById('circle-empty-illustration').style.display = 'block';
 		circleContainer.style.display = 'none';
-		document.getElementById('info-container').innerHTML = `You haven't visited any monetized websites yet! What are you waiting for? Get out there and explore the wild wild web.<br>Not sure where to start? <a href="https://coil.com/explore" target="_blank">Discover some monetized sites!</a>`;
+		setContentOfEl(document.getElementById('info-container'), [
+			textEl(`You haven't visited any monetized websites yet! What are you waiting for? Get out there and explore the wild wild web.`),
+			linebreakEl(),
+			textEl(`Not sure where to start? `),
+			linkEl('https://coil.com/explore', 'Discover some monetized sites!')
```
Comment (sharon-wang) [inline src/popup/popup.js]: nit: use backticks for strings (instead of single quote)

File: src/popup/popup.js [hunk 2]
```diff
@@ -74,38 +74,53 @@ async function getStats() {
...
 				}
 			} else {
 				// No payment has been streamed to the site -- present estimation based on Coil rate
 				const estimatedPaymentSentInUSD = getEstimatedPaymentForTimeInUSD(originStats.totalMonetizedTimeSpent);
 				if (estimatedPaymentSentInUSD > 0) {
-					document.getElementById('monetized-sent-text').innerHTML = `If you were using <a href="https://www.coil.com/">Coil</a> you would have sent `;
-					document.getElementById('monetized-sent-data').innerHTML = `<strong>$${estimatedPaymentSentInUSD}<span style="font-size: 12px;">USD</span></strong>.`;
+					setContentOfEl(document.getElementById('monetized-sent-text'), [
+						textEl(`If you were using `),
+						linkEl('https://www.coil.com/', 'Coil'),
```
Comment (sharon-wang) [inline src/popup/popup.js]: same nit: single quote --> backtick

File: src/popup/popup.js [hunk 3]
```diff
@@ -74,38 +74,53 @@ async function getStats() {
...
+						textEl(` you would have sent `),
+					]);
+					setContentOfEl(document.getElementById('monetized-sent-data'), [
+						setContentOfEl(strongEl(), [
+							textEl(`$${estimatedPaymentSentInUSD}`),
+							smallTextEl('USD')
```
Comment (sharon-wang) [inline src/popup/popup.js]: same nit: single quote --> backtick

File: src/popup/popup.js [hunk 4]
```diff
@@ -74,38 +74,53 @@ async function getStats() {
...
 			}
 
 			const monetizedTimePercent = getMonetizedTimeSpentPercent(originStats);
 			if (monetizedTimePercent) {
-				document.getElementById('monetized-percent-data').innerHTML = monetizedTimePercent+'%';
+				document.getElementById('monetized-percent-data').innerText = `${monetizedTimePercent}%`;
 			} else {
 				// TODO: ADD A CHANGE OF TEXT!
-				document.getElementById('monetized-percent-data').innerHTML = '0%';
+				document.getElementById('monetized-percent-data').innerText = '0%';
```
Comment (sharon-wang) [inline src/popup/popup.js]: same nit: single quote --> backtick

File: src/popup/popup.js [hunk 5]
```diff
@@ -136,15 +151,15 @@ async function getStats() {
...
+						linkEl.innerText = originData.origin.replace(URL_PREFIX_REGEX, "");
 
 						needsLoveContainer.appendChild(linkEl);
 						const brEl = document.createElement('br');
 						needsLoveContainer.appendChild(brEl);
 					}
 				} else {
 					const el = document.createElement('span');
-					el.innerHTML = 'No sites visited yet!';
+					el.innerText = 'No sites visited yet!';
```
Comment (sharon-wang) [inline src/popup/popup.js]: same nit: single quote --> backtick

File: src/popup/popup.js [hunk 6]
```diff
@@ -289,53 +317,134 @@ function createTopSiteDetailHTML(originData, originStats) {
...
+				textEl(`You've visited `),
+				strongEl(visitCountText),
+				textEl(`, which is `),
+				strongEl(`${percentVisits}%`),
+				textEl(` of your total website visits.`)
+			]);
 		} else {
-			visitCountString = `Counting up visits...<br><br>`;
+			setContentOfEl(visitCountEl, [
+				textEl('Counting up visits...')
```
Comment (sharon-wang) [inline src/popup/popup.js]: same nit: single quote --> backtick

File: src/popup/popup.js [hunk 7]
```diff
@@ -289,53 +317,134 @@ function createTopSiteDetailHTML(originData, originStats) {
...
 
-	return `<a href="${origin}" style="color: black; text-decoration: underline;">${origin}</a><br><br>
-		${timeSpentString}
-		${visitCountString}
-		${paymentString}`;
+	return [linkEl(origin, origin), timeSpentEl, visitCountEl, paymentEl];
+	// return `<a href="${origin}" style="color: black; text-decoration: underline;">${origin}</a><br><br>
+	// 	${timeSpentString}
+	// 	${visitCountString}
+	// 	${paymentString}`;
```
Comment (sharon-wang) [inline src/popup/popup.js]: pls yeet this

File: src/popup/popup.js [hunk 8]
```diff
@@ -289,53 +317,134 @@ function createTopSiteDetailHTML(originData, originStats) {
...
-	topSiteDetailEl.innerHTML = innerHTML;
+	setContentOfEl(topSiteDetailEl, els);
+}
+
+/**
+ * Convert the provided sentAssetsMap to list of HTML elements.
+ * Example output: [
+ * 	<strong>20.20<span style="font-size: 12px;">CAD</span></strong>,
+ * 	<strong>0.67<span style="font-size: 12px;">XRP</span></strong>,
+ *  textEl(` and `),
```
Comment (sharon-wang) [inline src/popup/popup.js]: maybe just ` and `?

File: src/popup/popup.js [hunk 9]
```diff
@@ -289,53 +317,134 @@ function createTopSiteDetailHTML(originData, originStats) {
...
+ *  <strong>123.45<span style="font-size: 12px;">USD</span></strong>
```
Comment (sharon-wang) [inline src/popup/popup.js]: nit: indentation is off here vs lines above

File: src/popup/popup.js [hunk 10]
```diff
@@ -398,12 +507,12 @@ function onSlideChange() {
 	if (currentSlide === carouselElements.length - 1) {
 		rightButtonEl.classList.add('disabled-button');
 
-		document.getElementById('intro-exit').innerHTML = 'done';
+		document.getElementById('intro-exit').innerText = 'done';
```
Comment (sharon-wang) [inline src/popup/popup.js]: same nit: single quote --> backtick

File: src/popup/popup.js [hunk 11]
```diff
@@ -398,12 +507,12 @@ function onSlideChange() {
...
 		document.getElementById('intro-exit').style.color = '#C31354';
 	} else {
 		rightButtonEl.classList.remove('disabled-button');
 
-		document.getElementById('intro-exit').innerHTML = 'skip';
+		document.getElementById('intro-exit').innerText = 'skip';
```
Comment (sharon-wang) [inline src/popup/popup.js]: same nit: single quote --> backtick

**Review by sharon-wang — COMMENTED** [review]
Just a bunch of nits, otherwise looks good!

Going to test on some browsers now.

**Comment by sharon-wang** [comment]
Looks good on Chrome, Brave, Edge. Will test on Firefox after your updates & rebase off master.

(testing without WM provider)

**Comment by sharon-wang** [comment]
Looks good on FF too. Ready for a squash and then merge :)

**Review by sharon-wang — COMMENTED** [review]
This was merged without explicit "Approval", but this is approved.

**Pull request merged**